### ISSUE TITLE
filtering has no effect for unweighted results when using a calibrated survey

### ISSUE THREAD

**Issue by vuwdanarcher** [gergness/srvyr#82]
Thank you for srvyr, it is such a brilliant package!

I'm finding that when using a calibrated survey, filtering does not work correctly for unweighted results. 

```
library(survey)
library(srvyr)

#create a calibrated survey design using survey package example
data(api)
dclus1<-svydesign(id=~dnum, weights=~pw, data=apiclus1, fpc=~fpc)
pop.totals<-c(`(Intercept)`=6194, stypeH=755, stypeM=1018)
dclus1g3c <- calibrate(dclus1, ~stype+api99, c(pop.totals, api99=3914069), calfun="raking")

#create a srvyr survey object for the non-calibrated survey
dclus1_survey <- as_survey(dclus1)

dclus1_survey %>%
  #filtering works fine for unweighted and survey_ratio
  filter(stype == "H") %>%
  summarise(target_n = unweighted(sum(target == 12, na.rm = TRUE)),
            target_N = unweighted(sum(!is.na(target), na.rm = TRUE)),
            target_pct = survey_ratio(target == 12, !is.na(target), na.rm = TRUE))

#create a srvyr object for the calibrated survey
dclus1g3c_survey <- as_survey(dclus1g3c)

dclus1g3c_survey %>%
  #filtering does not apply correctly to unweighted function, though survey_ratio works fine
  filter(stype == "H") %>%
  summarise(target_n = unweighted(sum(target == 12, na.rm = TRUE)),
            target_N = unweighted(sum(!is.na(target), na.rm = TRUE)),
            target_pct = survey_ratio(target == 12, !is.na(target), na.rm = TRUE))
```

commenting in/out the filter line for the calibrated survey has no effect. I'd really appreciate your help on this.

**Comment by bschneidr** [comment]
This is closely related to issue #51. The thread for that issue includes an explanation for why this behavior would occur. The short explanation is that for calibrated designs, subsetting the data doesn't actually drop rows from the data; instead, weights for such rows are effectively set to zero (technically, I think they're set to `Inf`, but that's interpreted as 0 for the purposes of calculations).

Perhaps the `unweighted()` function could be adjusted to look at the weights for calibrated designs to determine which rows are meant to be dropped due to filtering.

**Comment by vuwdanarcher** [comment]
Thanks for clarifying. I got around this by including some logic in the unweighted(sum()) command. It's not ideal but it will work for now.

```
dclus1g3c_survey %>%
  #filtering does not apply correctly to unweighted function, though survey_ratio works fine
  filter(stype == "H") %>%
  summarise(target_n = unweighted(sum(target == 12 & stype == "H", na.rm = TRUE)),
            target_N = unweighted(sum(!is.na(target) & stype == "H", na.rm = TRUE)),
            target_pct = survey_ratio(target == 12, !is.na(target), na.rm = TRUE))
```

**Comment by gergness** [comment]
closed via #83
